js-apps/meep-frontend/src/js/containers/meep-container.js

```diff
@@ -127,6 +143,42 @@ class MeepContainer extends Component {
     this.monitorTabFocus();
   }
 
+  componentWillMount() {
+    if (this.props.signInStatus === STATUS_SIGNIN_UNKNOWN) {
+      this.meepAuthApi.loginSupported((_, __, response) => {
+        if (response.status === 404) {
+          this.props.changeSignInStatus(STATUS_SIGNIN_NOT_SUPPORTED);
+          this.props.changeCurrentPage(PAGE_CONFIGURE);
+          this.props.changeTabIndex(PAGE_CONFIGURE_INDEX);
+        } else if (response.status === 200) {
+          this.props.changeSignInStatus(STATUS_SIGNED_IN);
+        } else {
+          this.props.changeSignInStatus(STATUS_SIGNED_OUT);
+          this.logout();
+        }
+      });
+    }
+
+    // Handle OAuth login in progress
+    if (this.props.signInStatus === STATUS_SIGNING_IN) {
+      let params = (new URL(document.location)).searchParams;
+      let userName = params.get('user');
+      if (userName) {
+        this.props.changeSignInUsername(userName);
+        // Remove Query params from URL now that we have stored them
+        window.history.replaceState({}, document.title, '/');
+        this.props.changeSignInStatus(STATUS_SIGNED_IN);
+        this.props.changeCurrentPage(PAGE_CONFIGURE);
+        this.props.changeTabIndex(PAGE_CONFIGURE_INDEX);
+        this.startSessionKeepaliveTimer();
+      } else {
+        // Sign in failed
+        this.logout();
+        this.props.changeSignInStatus(STATUS_SIGNED_OUT);
+      }
+    }
+  }
+  
   // Timers
   startTimers() {
```

Comment:
Should the timers take the sign-in status into account? For example, if we are signed out then we should not poll the platform pod states, etc.

We can either stop the timers or add a sign-in state check while running the periodic functions to avoid operations that we know will not work.

Reply:
Added check whether user is signed in or login not supported before starting timers